Content.Shared/Silicons/StationAi/SharedStationAiSystem.Held.cs

```diff
@@ -107,42 +108,67 @@ private void OnMessageAttempt(BoundUserInterfaceMessageAttempt ev)
         if (ev.Actor == ev.Target)
             return;
 
+        StationAiWhitelistComponent? whitelistComponent = null;
         if (TryComp(ev.Actor, out StationAiHeldComponent? aiComp) &&
            (!ValidateAi((ev.Actor, aiComp)) ||
-            !HasComp<StationAiWhitelistComponent>(ev.Target)))
+            !TryComp(ev.Target, out whitelistComponent)))
         {
             ev.Cancel();
```

Comment:
Just inline the component no reason to leave it separate.

Reply:
you mean - use one block?  and always call ShowDeviceNotRespondingPopup? this doesn't sound right - popup is kinda bound to whitelistComponent is { Enabled: false } scenario - there is no reason to show it if ai is not valid or we have no StationAiHeldComponent comp

Reply:
I cannot inline declaration because then declaration (which is required in the next lines) will be optional (based on previous conditions eval) and code won't compile :(

Reply:
And i don't want popup spam on every click! :/ so i want popup in separate condition... otherwise it will spam popup on every click!

Reply:
It absolutely will compile, the only reason it won't is if you don't get the component out of it, just adjust the if statement for it.